docs/en/ingest-management/fleet/add-fleet-server.asciidoc

```diff
@@ -9,6 +9,12 @@ when deploying {fleet-server}:
 * <<deployed-on-prem>> to work with {es} running on-premises. You self-manage both {fleet-server} and {es}.
 * <<fleet-server-on-prem-es-cloud>>. You manage {fleet-server} and Elastic manages {es}.
 
+ifeval::["{serverless-feature-flag}"=="yes"]
+IMPORTANT: On-premises {fleet-server} is not currently available for use with
+link:{serverless-docs}[{serverless-full}] projects. In a {serverless-short}
+environment we recommend using {fleet-server} hosted on {ess}.
+endif::[]
```

Comment:
> This text makes sense for some serverless solutions, but not ES3.

Thanks. I should have said "Elastic Cloud" rather than "Elasticsearch Service", so maybe that text is better? I've updated the screen capture with that change.

> Could we (and should we) use feature flags to display custom text for each solution?

We can certainly have different text in, say, the Observability Guide, versus the Security Guide, versus the Fleet & Agent Guide, and other books.